agents/fts_stage/fts_agent.py

```diff
@@ -0,0 +1,304 @@
+
+    def close(self):
+        self.comm.close()
+
+    
+
+class FTSMirrorAgent:
+    """
+    Agent for connecting to the FTS mirror control
+
+    Args:
+        ip_addr: IP address of Motion Controller
+        port: Port of Motion Controller
+        mode: 'acq': Start data acquisition on initialize
+        samp: default sampling frequency in Hz
+ 
+    """
+
+    def __init__(self, agent, ip_addr, port, mode=None, samp=2):
+
+        self.ip_addr = ip_addr
+        self.port = int(port)
+
+        self.stage = None
+        self.initialized = False
+        self.take_data = False
+
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        if mode == 'acq':
+            self.auto_acq = True
+        else:
+            self.auto_acq = False
+        self.sampling_frequency = float(samp)
+
+        ### register the position feeds
+        agg_params = {
+            'frame_length' : 10*60, #[sec]
+        }
+
+        self.agent.register_feed('position',
+                                 record = True,
+                                 agg_params = agg_params,
+                                 buffer_time = 0)
+
+    def init_stage_task(self, session, params=None):
+        """init_stage_task(params=None)
+        Perform first time setup for communication with FTS stage.
+
+        Args:
+            params (dict): Parameters dictionary for passing parameters to
+                task.
+        """
+
+        if params is None:
+            params = {}
+        
+        if self.stage is not None and self.initialized:
+            return True, 'Stages already Initialized'
+           
+        self.log.debug("Trying to acquire lock")
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already" \
+                                "running".format(self.lock.job))
+                return False, "Could not acquire lock."
+            # Run the function you want to run
+            self.log.debug("Lock Acquired Connecting to Stages")
+            self.stage = FTSStage(self.ip_addr, self.port)
+
+        # This part is for the record and to allow future calls to proceed,
+        # so does not require the lock
+        self.initialized = True
+        if self.auto_acq:
+            self.agent.start('acq')
+        return True, 'Stage Initialized.'
+
+    def home_task(self, session, params=None):
+        """ Home the stage to its negative limit
+        """
+
+        with self.lock.acquire_timeout(timeout=3, job='home') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start home because lock held by" \
+                                "{self.lock.job}")
+                return False, "Could not get lock"
+            self.stage.home()
```

Comment:
What's the response read after sending the `HOME X` command? Should it checked here to make sure it was successful?

Reply:
It returns nothing useful. The only thing is the read call blocks until the home is done. And since the controller ignores everything while it's running I have that read call there mostly just to block.